### DISCUSSION TITLE
Using ports on GitHub Codespaces?

### DISCUSSION THREAD

**Discussion by panicboat** [argoproj/argo-workflows#12655] [Q&A] [answered]
Opening Codespaces and accessing port 42385 via HTTPS results in "Client sent an HTTP request to an HTTPS server."
How can I resolve this?

**Comment by Joibel** [comment]
When does this happen? I can open in code spaces without this error message.

**Reply by panicboat** [reply]
Thanks for confirming.
This happens after I run "Create codespace".

I see the following log in terminal.
```console
2024/02/11 14:58:33 install: process exited 27664: <nil>
2024/02/11 14:58:33 install: starting downstream tasks
2024/02/11 14:58:33 install: starting: example
example: waiting for mutex "docker"
example: locked mutex "docker"
2024/02/11 14:58:33 example: starting process
2024/02/11 14:58:33 example: resetting process
2024/02/11 14:58:33 example: running process
2024/02/11 14:58:33 example: starting process "kubectl create -f examples/hello-world.yaml"
2024/02/11 14:58:33 install: context cancelled, stopping process
2024/02/11 14:58:33 install: stopping process 27664
2024/02/11 14:58:33 install: finding process 27664
2024/02/11 14:58:33 install: terminating process 27664
2024/02/11 14:58:33 example: started process "kubectl create -f examples/hello-world.yaml": <nil>
2024/02/11 14:58:33 example: getting pgid for 28251
2024/02/11 14:58:33 example: pgid for 28251 is 28251 <nil>
2024/02/11 14:58:33 example: waiting for process 28251 pgid 28251 ("kubectl create -f examples/hello-world.yaml")
2024/02/11 14:58:33 install: waiting 3s before killing 27664
example: workflow.argoproj.io/hello-world-cr2lp created
2024/02/11 14:58:34 example: process exited 28251: <nil>
2024/02/11 14:58:34 example: starting downstream tasks
2024/02/11 14:58:34 example: starting: pre-up
2024/02/11 14:58:34 pre-up: starting process
2024/02/11 14:58:34 example: context cancelled, stopping process
2024/02/11 14:58:34 example: stopping process 28251
2024/02/11 14:58:34 example: finding process 28251
2024/02/11 14:58:34 pre-up: resetting process
2024/02/11 14:58:34 pre-up: running process
2024/02/11 14:58:34 pre-up: starting downstream tasks
2024/02/11 14:58:34 example: terminating process 28251
2024/02/11 14:58:34 example: waiting 3s before killing 28251
+ kubectl get cm coredns -n kube-system -o yaml
+ kubectl apply -f -
+ grep host.docker.internal /etc/hosts
+ sed s/  NodeHosts: |/  NodeHosts: |\n    172.17.0.1   host.docker.internal/
Warning: resource configmaps/coredns is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
configmap/coredns configured
Outcome: success User: vscode WorkspaceFolder: /home/vscode/go/src/github.com/argoproj/argo-workflows
devcontainer process exited with exit code 0
Finished configuring codespace.
```
Then when I try to open the "Port" tab and open the address for automatic forwarding, I get this message.
![image](https://github.com/argoproj/argo-workflows/assets/4182311/43490b2b-b316-4c46-95f9-a612f2d335ad)
![image](https://github.com/argoproj/argo-workflows/assets/4182311/66caa9f1-7b7d-46f4-b00e-ee234f257439)
Am I missing some procedure?

**Reply by Joibel** [reply]
I am now more confused. In your first message you state an error on port 42385 regarding HTTP and HTTPS.
The information you've then followed up with does not contain any error messages like that, and appears to be unrelated.
I still don't understand where you are seeing a problem here.

**Reply by panicboat** [reply]
Sorry for the confusion.
The 42385 and 35835 ports appear to be ports that change each time they are activated.

Let me get a few things straight.

### Steps to reproduce

1. Create codespace
2. Accessing URLs published in codespaces

### phenomenon

At this time, the browser displays the message "Client sent an HTTP request to an HTTPS server."
Also, the developer tools had an error message like the aforementioned screenshot

**Reply by agilgur5** [reply, marked as answer]
Per your screenshot, the specific port you're accessing seems to be the `docker-proxy`. I'm not sure why you're trying to access that. That seems like it might be Codespaces's as well, and not Argo's.

You usually only want the Server or the UI ports, which, [per the docs](https://argo-workflows.readthedocs.io/en/latest/running-locally/#developing-locally), are `2746` and `8080`

**Reply by panicboat** [reply]
It seems that the port to be accessed was wrong.
The UI was displayed by accessing port 8080 after `make start UI=true`.
Sorry for the fuss, thanks for the help.
